be/src/exprs/array_functions.h

```diff
@@ -70,27 +70,20 @@ class ArrayFunctions {
 
     DEFINE_VECTORIZED_FN(array_slice);
 
-#define DEFINE_ARRAY_OVERLAP_FN(NAME, LT)                                                               \
-    static StatusOr<ColumnPtr> array_overlap_##NAME(FunctionContext* context, const Columns& columns) { \
-        return ArrayOverlap<LT>::process(context, columns);                                             \
+    template <LogicalType type>
```

Comment:
suggest to use class template and add a guard to template, so it we can provide different specializations for different types in the future possibly.

for an exmaple:

```
template<LogicalType type, typename = Guard>
struct ArrayIntersectFunction {
static ColumnPtr evaluate(const Columns& args){
}
}；

template<LogicalType type>
struct ArrayIntersectFunction<type, DecimalTypeGuard<type>> {
static ColumnPtr evaluate(const Columns& args){
}
}；
```

Reply:
ok, here just refactor to template, will add guard in next pr to support decimal